### ISSUE TITLE
Erasure: scala.Array survives erasure.

### ISSUE THREAD

**Issue by odersky** [lampepfl/dotty#996]
Compiling

    ./scala-scala/src/library/scala/collection/immutable/TrieIterator.scala

gives a crash in the backend. Everything -Ychecks OK before.

java.lang.AssertionError: assertion failed: class Array
	at scala.Predef$.assert(Predef.scala:165)
	at dotty.tools.backend.jvm.DottyBackendInterface$$anon$10.primitiveOrClassToBType$1(DottyBackendInterface.scala:814)
	at dotty.tools.backend.jvm.DottyBackendInterface$$anon$10.toTypeKind(DottyBackendInterface.scala:835)
	at scala.tools.nsc.backend.jvm.BCodeHelpers$BCInnerClassGen$class.toTypeKind(BCodeHelpers.scala:272)
	at scala.tools.nsc.backend.jvm.BCodeSkelBuilder$PlainSkelBuilder.toTypeKind(BCodeSkelBuilder.scala:51)
	at scala.tools.nsc.backend.jvm.BCodeSkelBuilder$PlainSkelBuilder.tpeTK(BCodeSkelBuilder.scala:84)
	at scala.tools.nsc.backend.jvm.BCodeBodyBuilder$PlainBodyBuilder.genTypeApply(BCodeBodyBuilder.scala:582)
	at scala.tools.nsc.backend.jvm.BCodeBodyBuilder$PlainBodyBuilder.genLoad(BCodeBodyBuilder.scala:429)

**Comment by odersky** [comment]
You need to pull the updated dotty-library, otherwise the file is rejected by Typer.

**Comment by DarkDimius** [comment]
@odersky, could you please help me reproduce the issue?
If I have dotty on c864e118e7bd04dcbd0503ab9af6729fda94ff3f and library on https://github.com/DarkDimius/scala/commit/7f048c42a619cc19ebf70de9f23dc68eea20537b 

than I get error from typer:

```
./scala-scala/src/library/scala/collection/immutable/TrieIterator.scala:69: error: covariant type T occurs in invariant position in type Array[scala.collection.immutable.Iterable[T]] of value xs
  final class DupIterator(xs: Array[Iterable[T]]) extends TrieIterator[T](xs,
                            ^
```

**Comment by odersky** [comment]
@darkdimius You need to compile with -language:Scala2. Then it passes the typer for me.

**Comment by DarkDimius** [comment]
Was able to reproduce.

**Comment by DarkDimius** [comment]
@odersky I traced this down to `scala.Array` surviving erasure, I've improved post-condition in erasure to check this: #1005

**Comment by DarkDimius** [comment]
fixed in #1005
